tests/density_estimator_test.py

```diff
@@ -264,3 +59,131 @@ def test_shape_handling_utility_for_density_estimator(
         f"Shapes of Output ({output.shape}) and target shape ({target_shape}) do not "
         f"match."
     )
+
+
+@pytest.mark.parametrize(
+    "density_estimator_name", ("mdn", "maf", "zuko_maf", "categorical", "mixed")
+)
+@pytest.mark.parametrize("input_iid_dim", (1, 2))
+@pytest.mark.parametrize("input_event_shape", ((1,), (4,)))
+@pytest.mark.parametrize("condition_event_shape", ((1,), (7,)))
```

Comment:
We should also include some tests that input images i.e. 2d  event shape or more.

This requires specifying an "fake" embedding net (see previous tests)